apps/system/test/marionette/task_manager_icons_test.js

```diff
@@ -0,0 +1,181 @@
+'use strict';
+
+var Server = require('../../../../shared/test/integration/server');
+var TaskManager = require('./lib/task_manager');
+var Rocketbar = require('./lib/rocketbar');
+var FakeApp = require('./lib/fakeapp');
+var assert = require('assert');
+
+marionette('Task Manager - Icons', function() {
+  var iconAppOrigin = 'iconapp.gaiamobile.org';
+  var apps = {};
+  apps[iconAppOrigin] = __dirname + '/../apps/icons-app';
+
+  var client = marionette.client({
+    profile: {
+      apps: apps
+    },
+    desiredCapabilities: { raisesAccessibilityExceptions: false }
+  });
+
+  var actions, server, system, rocketbar, taskManager;
+  var firefoxApp;
+
+  suiteSetup(function(done) {
+    Server.create(__dirname + '/../apps/icons-app/', function(err, _server) {
+      server = _server;
+      done();
+    });
+  });
+
+  suiteTeardown(function() {
+    server.stop();
+  });
+
+  // using structured art (known colors for each icon image)
+  // we can map from a color to a particular size icon image
+  var iconColorToSize = {
+    '255,255,0,255': 16,
+    '0,255,255,255': 32,
+    '255,0,255,255': 64,
+    '0,0,255,255': 128,
+    '255,0,0,255': 256
+  };
+
+  function getElementColor(elem, expectedSize) {
```

Comment:
badass :)
Can you file a bug to investigate using this to cover the screenshoting in the card view?